mail_environment/models/fetchmail_server.py

```diff
@@ -3,58 +3,46 @@
 
 import operator
 from odoo import api, fields, models
-from odoo.addons.server_environment import serv_config
 
 
 class FetchmailServer(models.Model):
     """Incoming POP/IMAP mail server account"""
-    _inherit = 'fetchmail.server'
-
-    server = fields.Char(compute='_compute_server_env',
-                         states={})
-    port = fields.Integer(compute='_compute_server_env',
-                          states={})
-    type = fields.Selection(compute='_compute_server_env',
-                            search='_search_type',
-                            states={})
-    user = fields.Char(compute='_compute_server_env',
-                       states={})
-    password = fields.Char(compute='_compute_server_env',
-                           states={})
-    is_ssl = fields.Boolean(compute='_compute_server_env')
-    attach = fields.Boolean(compute='_compute_server_env')
-    original = fields.Boolean(compute='_compute_server_env')
-
-    @api.depends()
-    def _compute_server_env(self):
-        for fetchmail in self:
-            global_section_name = 'incoming_mail'
-
-            key_types = {'port': int,
-                         'is_ssl': lambda a: bool(int(a or 0)),
-                         'attach': lambda a: bool(int(a or 0)),
-                         'original': lambda a: bool(int(a or 0)),
-                         }
-
-            # default vals
-            config_vals = {'port': 993,
-                           'is_ssl': 0,
-                           'attach': 0,
-                           'original': 0,
-                           }
-            if serv_config.has_section(global_section_name):
-                config_vals.update(serv_config.items(global_section_name))
+    _name = 'fetchmail.server'
+    _inherit = ["fetchmail.server", "server.env.mixin"]
+
+    @property
+    def _server_env_fields(self):
+        base_fields = super()._server_env_fields
+        mail_fields = {
+            "server": {},
+            "port": {
+                "getter": "getint",
+            },
+            "type": {},
+            "user": {},
+            "password": {},
+            "is_ssl": {
+                "getter": "getbool",
```

Comment:
could this be inferred from the field type?

Reply:
of course!

Reply:
done in 303e5e6

Reply:
cool, thanks!